src/corecel/sys/Stream.hh

```diff
@@ -56,8 +103,13 @@ class Stream
     // Access the stream
     StreamT get() const { return stream_; }
 
+    // Access the thrust resource allocator associated with the stream
+    ResourceT& get_memory_resource() const { return memory_resource_; }
```

Comment:
In celeritas we avoid `get_` prefix for simple accessors: maybe just `memory_resource`? If the `Stream` is going to encapsulate more than just a CUDA stream (which I think is a good idea) perhaps we should rename `get()` above to `stream()`? I also think we should make this function non-const, remove `mutable` below, and let the device return a mutable stream, if we're going to store temporary device buffers as part of it.